### PR TITLE
Implemented OneWeekView and TwoWeeksView

### LINKED ISSUE #23: I create another branch,Added Week View
git url is

https://github.com/MackYang/Xamarin.Plugin.Calendar/tree/AddWeekView

### PR DESCRIPTION
**Pull request by tborozan**
New concept **LayoutUnit**:
Represents a week, two weeks or month depending on `CalendarLayout` property

Added:
- CalendarPlugin
    - `CalendarLayout` - indicates selected layout
    - `WeekViewUnit` - represents **LayoutUnit** naming scheme (month name/week number)
    - `ViewLayoutEngine` - generated layout based on `CalendarLayout`
- SampleApp
    - `OneWeekView`
    - `TwoWeeksView`

Renamed:
- `MonthYear` -> `ShownDate`
- `ShowMonthPicker` -> `ShowLayoutUnitPicker`
- `MonthText` -> `LayoutUnitText`
- `PrevMonthCommand` -> `PrevLayoutUnitCommand`
- `NextMonthCommand` -> `PrevLayoutUnitCommand`

### CODE REVIEW

File: src/Calendar.Plugin.Sample/SampleApp/ViewModels/AdvancedPageViewModel.cs [hunk 1]
```diff
@@ -46,36 +47,34 @@ public AdvancedPageViewModel() : base()
...
-                    // insert/add items to observable collection
-                    todayEvents.Insert(0, new AdvancedEventModel { Name = "Cool event insert", Description = "This is Cool event's description!", Starting = new DateTime() });
-                    todayEvents.Add(new AdvancedEventModel { Name = "Cool event add", Description = "This is Cool event's description!", Starting = new DateTime() });
+                   // get observable collection later
+                   var todayEvents = Events[DateTime.Now] as ObservableCollection<AdvancedEventModel>;
 
-                }, TaskScheduler.FromCurrentSynchronizationContext());
-            }, TaskScheduler.FromCurrentSynchronizationContext());
+                   // insert/add items to observable collection
+                   todayEvents.Insert(0, new AdvancedEventModel { Name = "Cool event insert", Description = "This is Cool event's description!", Starting = new DateTime() });
```
Comment (m-luketin) [inline src/Calendar.Plugin.Sample/SampleApp/ViewModels/AdvancedPageViewModel.cs:74]: Might be good to name these consistently with the other events. We currently have Cool eventX, Cool event insert and Cool event add, its starting to look a bit messy

**Review by m-luketin — COMMENTED** [review]
When entering the single and two week view calendar sample, there is a snappy change a few seconds after the page seems fully loaded, and then another one after a second or so. It's confusing for the user opening the sample, especially if they attempt to tap anything.

Its unclear why some events have a margin, while others dont.

File: src/Calendar.Plugin.Sample/SampleApp/ViewModels/AdvancedPageViewModel.cs [hunk 2]
```diff
@@ -129,5 +138,31 @@ private async Task ExecuteEventSelectedCommand(object item)
                 await App.Current.MainPage.DisplayAlert(title, message, "Ok");
             }
         }
+
+        private void ChangeShownUnit(int amountToAdd)
+        {
+            switch (CalendarLayout)
```
Comment (m-luketin) [inline src/Calendar.Plugin.Sample/SampleApp/ViewModels/AdvancedPageViewModel.cs:144]: I think most of this is excess logic. Could we not just have a ChangeShown method (the same as ChangeShownWeek, where we send the proper parameter for changing the month aswell?

File: src/Calendar.Plugin/Shared/Controls/ViewLayoutEngines/MonthViewEngine.cs [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+        {
+            var grid = GenerateWeekLayout(
+                dayViews,
+                daysTitleHeight,
+                daysTitleColor,
+                daysTitleLabelStyle,
+                dayViewSize,
+                dayTappedCommand,
+                dayModelPropertyChanged,
+                6
```
Comment (m-luketin) [inline src/Calendar.Plugin/Shared/Controls/ViewLayoutEngines/MonthViewEngine.cs:36]: Separate magic number into a named private variable

File: src/Calendar.Plugin/Shared/Controls/ViewLayoutEngines/ViewLayoutBase.cs [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+            dayViews.Clear();
+
+            for (int i = 1; i <= numberOfWeeks; i++)
+            {
+                grid.RowDefinitions.Add(new RowDefinition()
+                {
+                    Height = dayViewSize,
+                });
+
+                for (int ii = 0; ii < 7; ii++)
```
Comment (dujesaric) [inline src/Calendar.Plugin/Shared/Controls/ViewLayoutEngines/ViewLayoutBase.cs:72]: Could we separate looping integers with constants? NumberOfDaysInWeek?

**Review by m-luketin — APPROVED** [review]
Everything seems in order. I just want to mention that, until now, days that don't belong to the selected month had the disabled color, which is not the case on the 1 and 2 week calendars. If this is intended then ok. (e.g. if we are on september, and the 1st and 2nd of october are visible, they would have the disabled color for the date number)

**Review by antonioseric — CHANGES_REQUESTED** [review]
something is wrong with event list when there is 5 or less events
Screenshot_1633279628

**Comment by inimirpaz** [comment]
> * `PrevMonthCommand` -> `PrevLayoutUnitCommand`
> 
> * `NextMonthCommand` -> `PrevLayoutUnitCommand`

The second one is actually `NextLayoutUnitCommand`

**Pull request merged**